core/key.go

```diff
@@ -318,6 +390,112 @@ func (k DataStoreKey) ToPrimaryDataStoreKey() PrimaryDataStoreKey {
 	}
 }
 
+// NewIndexDataStoreKey creates a new IndexDataStoreKey from a string.
+// It expects the input string is in the following format:
+//
+// /[CollectionID]/[IndexID]/[FieldValue](/[FieldValue]...)
+//
+// Where [CollectionID] and [IndexID] are integers
+func NewIndexDataStoreKey(key string) (IndexDataStoreKey, error) {
+	if key == "" {
+		return IndexDataStoreKey{}, ErrEmptyKey
+	}
+
+	if !strings.HasPrefix(key, "/") {
+		return IndexDataStoreKey{}, ErrInvalidKey
+	}
+
+	elements := strings.Split(key[1:], "/")
+
+	// With less than 3 elements, we know it's an invalid key
+	if len(elements) < 3 {
+		return IndexDataStoreKey{}, ErrInvalidKey
+	}
+
+	colID, err := strconv.Atoi(elements[0])
+	if err != nil {
+		return IndexDataStoreKey{}, ErrInvalidKey
+	}
+
+	indexKey := IndexDataStoreKey{CollectionID: uint32(colID)}
+
+	indID, err := strconv.Atoi(elements[1])
+	if err != nil {
+		return IndexDataStoreKey{}, ErrInvalidKey
+	}
+	indexKey.IndexID = uint32(indID)
+
+	for i := 2; i < len(elements); i++ {
```

Comment:
todo: A comment here would be nice to quickly highlight why we are starting from `i := 2`. Is easy to see why now, but if someone removes the above elements [0] and [1] or adds new ones to the logic, the comment will be helpful to insure they update the index.